pkg/healthcheck/healthcheck.go

```diff
@@ -1022,6 +1047,46 @@ func (hc *HealthChecker) checkLinkerdConfigConfigMap() (string, *configPb.All, e
 	return string(cm.GetUID()), configPB, nil
 }
 
+func (hc *HealthChecker) checkIssuerCertsNotExpiringTooSoon() error {
+	if time.Now().AddDate(0, 0, ExpirationWarningThresholdInDays).After(hc.issuerCerts.Certificate.NotAfter) {
+		return fmt.Errorf("certificate is expiring sooner than %d days from now", ExpirationWarningThresholdInDays)
+	}
+	return nil
+}
+
+func (hc *HealthChecker) checkIssuerCertsValidity() (*tls.Cred, error) {
+	_, configPB, err := FetchLinkerdConfigMap(hc.kubeAPI, hc.ControlPlaneNamespace)
+	if err != nil {
+		return nil, err
+	}
+
+	idctx := configPB.Global.IdentityContext
+	var data *issuercerts.IssuerCertData
+
+	if idctx.Scheme == k8s.IdentityIssuerSchemeLinkerd {
```

Comment:
if the scheme property is empty (which will be the case for previous linkerd installations) we should default to assuming the scheme is Linkerd.  E.g. when running `check` on my stable-2.6 cluster I get this error:

```
× issuer certs are valid
    key ca.crt containing the trust anchors needs to exist in secret linkerd-identity-issuer if --identity-external-issuer=true
    see https://linkerd.io/checks/#l5d-issuer-certs-are-valid for hints
```

because I don't have a scheme and we're falling through to the else block here.

Reply:
Ahh, very good catch, thanks!

Reply:
Added a test for that as well